components/navbar.js

```diff
@@ -0,0 +1,208 @@
+import * as React from "react";
+import AppBar from "@mui/material/AppBar";
+import Box from "@mui/material/Box";
+import Toolbar from "@mui/material/Toolbar";
+import IconButton from "@mui/material/IconButton";
+import Typography from "@mui/material/Typography";
+import Menu from "@mui/material/Menu";
+import MenuIcon from "@mui/icons-material/Menu";
+import Container from "@mui/material/Container";
+import Avatar from "@mui/material/Avatar";
+import Button from "@mui/material/Button";
+import Tooltip from "@mui/material/Tooltip";
+import MenuItem from "@mui/material/MenuItem";
+import { useOperation, useOperationMethod } from "react-openapi-client";
+import SchoolIcon from "@mui/icons-material/School";
+import axios from "axios";
+
+const pages = [{ text: "Home", url: "/" }];
+
+function ResponsiveAppBar() {
+  const [anchorElNav, setAnchorElNav] = React.useState(null);
+  const [anchorElUser, setAnchorElUser] = React.useState(null);
+  const [endSession] = useOperationMethod("endSession");
+  const { loading, data, error } = useOperation("getSession");
+
+  let userName = null;
+  let userId = 0;
+  let settings = {};
+
+  const HandleSingOut = async () => {
+    await endSession();
+    if (window.location.pathname === "/") {
+      window.location.reload();
+    } else {
+      axios.get("/");
+    }
+  };
+
+  const handleOpenNavMenu = (event) => {
+    setAnchorElNav(event.currentTarget);
+  };
+  const handleOpenUserMenu = (event) => {
+    setAnchorElUser(event.currentTarget);
+  };
+
+  const handleCloseNavMenu = () => {
+    setAnchorElNav(null);
+  };
+
+  const handleCloseUserMenu = () => {
+    setAnchorElUser(null);
+  };
+
+  if (loading) {
+    return <div>Loading...</div>;
```

Comment:
You could use the CircularProgress component from material, a little bit prettier.

Reply:
Thanks!! I didn't know that haha